tools/exp/main.go

```diff
@@ -0,0 +1,131 @@
+	errSettingUpOutputDir = "error setting up output directory: %s"
+
+	// errOutputDirNotDir is the error that is returned when the output directory is not a directory.
+	errOutputDirNotDir = errors.New("output directory is not a directory")
+
+	// errSettingUpEnv is the error returned when the environment variables could not be set up.
+	errSettingUpEnv = "error setting up environment variables: %s"
+
+	// errSettingUpClient is the error returned when the client could not be set up.
+	errSettingUpClient = "error setting up client: %s"
+
+	// errGeneratingFiles is the error returned when the files could not be generated.
+	errGeneratingFiles = "error generating files: %s"
+)
+
+const (
+	// appStarted is the message that is printed when the application starts.
+	appStarted = "exp tool started"
+
+	// settingUpOutputDir is the message that is printed when the output directory is being set up.
+	settingUpOutputDir = "setting up output directory"
+
+	// settingUpEnv is the message that is printed when the environment variables are being set up.
+	settingUpEnv = "setting up environment variables"
+
+	// settingUpClient is the message that is printed when the client is being set up.
+	settingUpClient = "setting up client"
+
+	// generatingFiles is the message that is printed when the files are being generated.
+	generatingFiles = "generating files"
+
+	// done is the message that is printed when the application is done.
+	done = "done"
+)
+
+// defaultOutputDir is the default output directory.
+const defaultOutputDir = "tools/exp/dist"
+
+// flags is the flags container of the application.
+var flags = &util.Flags{}
+
+// logger is the logger of the application.
+var logger = &util.Logger{}
+
+// env is a map of environment variables.
+var env = util.EnvMap{
+	util.EnvAivenProjectName: "",
+}
+
+// client is a pointer to the Aiven client.
+var client = &aiven.Client{}
+
+// setup sets up the application.
+func setup() {
+	util.SetupFlags(flags)
+
+	util.SetupLogger(logger)
+
+	logger.Info.Println(appStarted)
+
+	logger.Info.Println(settingUpOutputDir)
+
+	if err := setupOutputDir(); err != nil {
+		logger.Error.Fatalf(errSettingUpOutputDir, err)
+	}
+
+	logger.Info.Println(settingUpEnv)
+
+	if err := util.SetupEnv(env); err != nil {
+		logger.Error.Fatalf(errSettingUpEnv, err)
+	}
+
+	logger.Info.Println(settingUpClient)
+
+	if err := util.SetupClient(client); err != nil {
+		logger.Error.Fatalf(errSettingUpClient, err)
+	}
+}
+
+// setupOutputDir sets up the output directory.
+func setupOutputDir() error {
+	if flags.OutputDir == "" {
+		wd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		flags.OutputDir = strings.Join([]string{wd, defaultOutputDir}, string(os.PathSeparator))
+	}
+
+	fi, err := os.Stat(flags.OutputDir)
+	if err != nil {
+		return err
+	}
+
+	if !fi.IsDir() {
+		return errOutputDirNotDir
+	}
+
+	return nil
+}
+
+// main is the entrypoint for the application.
+func main() {
```

Comment:
Can you add interruption signal handling?

Reply:
will do